test/FunctionalTests/Infrastructure/ApplicationTestFixture.cs

```diff
@@ -67,56 +61,29 @@ public static DeploymentParameters GetDeploymentParameters(string applicationPat
             return deploymentParameters;
         }
 
-        public virtual ILoggerFactory CreateLoggerFactory()
-        {
-            return new LoggerFactory().AddConsole();
-        }
-
         public void Dispose()
         {
-        }
-
-        private static void TryDeleteDirectory(string directory)
-        {
-            try
+            if (_deploymentTask?.Status == TaskStatus.RanToCompletion)
             {
-                Directory.Delete(directory, recursive: true);
+                _deploymentTask.Result.HttpClient?.Dispose();
             }
-            catch (IOException)
-            {
-                // Ignore delete failures.
-            }
-        }
 
-        public async Task<Deployment> CreateDeploymentAsync(RuntimeFlavor flavor)
-        {
-            var deploymentParameters = GetDeploymentParameters(flavor);
-            var deployer = ApplicationDeployerFactory.Create(deploymentParameters, LoggerFactory);
-            var deploymentResult = await deployer.DeployAsync();
-
-            return new Deployment(deployer, deploymentResult);
+            _deployer?.Dispose();
         }
 
-        public class Deployment : IDisposable
+        public async Task<DeploymentResult> CreateDeploymentAsync(ILoggerFactory loggerFactory)
         {
-            public Deployment(IApplicationDeployer deployer, DeploymentResult deploymentResult)
+            lock (_deploymentLock)
```

Comment:
It does However, we want the deployment to have access to the test logger (`ITestOutputHelper`) and that's not available until the test ctor:

```
C#
public class Test : IClassFixture<Fixture>
{
   public Test(Fixture f, ITestOutputHelper helper) // No way to pass it in to Fixture at the time it's constructed (or so it seems).
}
```

We defer the deployment until the actual test so we can pass the logger in. Plus it also means we don't risk calling the virtual `GetDeploymentParameters` from the fixture's ctor.